pandas/core/indexing.py

```diff
@@ -2222,6 +2224,12 @@ def check_bool_indexer(index: Index, key) -> np.ndarray:
                 "the indexed object do not match)."
             )
         result = result.astype(bool)._values
+    elif is_extension_array_dtype(key) and is_bool_dtype(key):
+        mask = isna(key)
```

Comment:
This should likely be `key.to_numpy(fill_value=False)`.

I think that will avoid the full scans of the values from the `isna` and `.any()`.

Reply:
Yes, that's much nicer